linux-create-vm.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+# Update and Upgrade apt
+
+sudo apt update
+sudo apt upgrade
+
+# Install git
+  echo "git should be installed"
+if ( which git )
+then 
+  echo "git already installed"
+else 
+  echo "installing git"
+  sudo apt install git
+fi
+
+#install nano
+  echo "nano should be installed"
+if ( which nano )
+then
+  echo "nano already installed"
+else
+  echo "installing nano"
+  sudo apt install nano 
+fi
+
+#install snap
+  echo "nano should be installed"
+if ( which snap )
+then
+  echo "snap already installed"
+else
+  echo "installing snap"
+  sudo apt install snap 
+fi
+
+# Install multipass
+  echo "installing multipass on $(uname)"
+if  ( which multipass )
+then
+  echo "multipass already installed on $(uname)"
+else
+  echo "installing multipass on $(uname)"  
+  sudo snap install multipass
+  sleep 5
+fi
+
+# Set Multipass local driver to Qemu
+if [ "$(multipass get local.driver)" = "qemu" ]
+then
+  echo -e "\n==== Qemu local driver set ====\n"
+else
+  echo -e "\n==== Setting Qemu as Multipass local driver ====\n"
+  multipass set local.driver=qemu
+fi
+
+# ssh keys
+if [ -f "./id_ed25519" ]
+#if [ stat "./id_ed25519" ]
+then
+  echo "trifecta ssh key exists"
+else
+  echo "creating ssh keys for trifecta"
+  ssh-keygen -f ./id_ed25519 -t ed25519 -b 4096 -N ''
+fi
+
+# Create cloud-init.yaml
+if [ -f cloud-init.yaml ] 
+then
+  echo -e "\n==== Cloud-init.yaml present ====\n"
+else
+  echo -e "\n==== Creating cloud-init.yaml ====\n"
+  cat <<- EOF > cloud-init.yaml
+users:
+  - default
+  - name: $USER
+    sudo: ALL=(ALL) NOPASSWD:ALL
+    shell: /bin/bash
+    ssh_authorized_keys:
+      - $(cat id_ed25519.pub)
+EOF
+fi
+
+# Spinning up an ubuntu vm
+if ( multipass info trifecta | grep Running )
+then 
+  echo "trifecta vm is running"
+else 
+  echo "launching a ubuntu vm named trifecta"
+  multipass launch --name trifecta --cloud-init cloud-init.yaml
+fi
+
+# Copies webserver.sh to VM
+scp -i ./id_ed25519 -o StrictHostKeyChecking=no webserver.sh aws-cli-install.sh ansible-install.sh jenkins-install.sh amazon-cli-install.sh docker-install.sh $USER@$(multipass info trifecta | grep IPv4 | awk '{ print $2 }'):~/
```

Comment:
I was copying all of these in the begining, but It got really confusing. We should just do `docker-install.sh, jenkins-install.sh`, because the docker is done, and the jenkins is currently being worked. We will add to the `scp` command as we complete the scripts.